### PR TITLE
ch4/{ofi,ucx}: Improved error checking for business card exchange

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Business card exchange in roots only mode involves communication over a special "init" communicator. Take care to check for errors that may arise during the exchange and skip the cleanup of the init comm in that case to avoid masking the source of the error. Fixes pmodels/mpich#7200.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by raffenet** [comment]
test:mpich/ch4/most

**Comment by raffenet** [comment]
test:mpich/ch4/most

**Comment by raffenet** [comment]
test:mpich/ch4/most

**Comment by raffenet** [comment]
test:mpich/ch4/most

**Comment by raffenet** [comment]
test:mpich/ch4/most

File: src/mpid/ch4/src/init_comm.c [hunk 1]
```diff
@@ -60,18 +60,24 @@ int MPIDI_create_init_comm(MPIR_Comm ** comm)
...
 {
+    int mpi_errno = MPI_SUCCESS;
     int in_use;
     MPIR_Comm *comm = NULL;
     if (*comm_ptr != NULL) {
         comm = *comm_ptr;
         MPIDIU_release_lut(MPIDI_COMM(comm, map).irreg.lut.t);
         MPIDIG_destroy_comm(comm);
+        /* wait for any outstanding requests to complete */
+        MPIDIU_PROGRESS_WHILE(MPIR_Object_get_ref(comm) > 1, 0);
```
Comment (raffenet) [inline src/mpid/ch4/src/init_comm.c:73]: I'm thinking of dropping this patch and just including the additional error checks. It's possible the assertion failure `MPIR_Assertp(in_use == 0);` observed on Aurora was just the result of a previously masked error. @hzhou what do you think?

Comment (hzhou) [reply src/mpid/ch4/src/init_comm.c:73]: I agree. There are not many nondeterministic things during init. When we destroy init_comm, all operations should be locally complete already, or it is likely a bug. How about skip this for now and confirm on Aurora?

**Comment by raffenet** [comment]
test:mpich/ch4/most

**Comment by colleeneb** [comment]
This is ok to try testing on Aurora now, correct?

**Comment by raffenet** [comment]
> This is ok to try testing on Aurora now, correct?

Yes, this branch is OK to test now.

**Review by hzhou — APPROVED** [review]
LGTM

**Pull request merged**